R/bmerge.R

```diff
@@ -92,6 +92,14 @@ bmerge = function(i, x, icols, xcols, roll, rollends, nomatch, mult, ops, verbos
       }
       stopf("Incompatible join types: %s (%s) and %s (%s). Factor columns must join to factor or character columns.", xname, x_merge_type, iname, i_merge_type)
     }
+    # data.table::as.ITime, chron::times, nanotime::nanotime
+    time_classes = c("Date", "POSIXt", "ITime", "times", "nanotime")
+    x_class_time = intersect(class(x[[xcol]]), time_classes)
+    i_class_time = intersect(class(i[[icol]]), time_classes)
+    if (length(x_class_time) > 0L && length(i_class_time) > 0L && !identical(x_class_time, i_class_time)) {
```

Comment:
I removed the `sort()` check of the classes -- I figure if someone's using some weird ordering of their S3 class, throwing a warning can't hurt. It may well be that having mismatched class ordering results in some unexpected behavior anyway if the "wrong" dispatch eventually happens.